src/komponenter/din-situasjon/situasjon.test.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import Situasjon from './situasjon';
+import { ComponentType, ReactChildren, ReactElement } from 'react';
+import merge from 'merge-deep';
+import { DeepPartial } from 'redux';
+
+import {
+    Data as BrukerRegistreringData,
+    initialState as brukerRegistreringInitialState,
+    BrukerregistreringContext,
+} from '../../ducks/brukerregistrering';
+import { Data as SituasjonData, initialState as situasjonInitialState, SituasjonContext } from '../../ducks/situasjon';
+import {
+    Data as FeatureToggleData,
+    initialState as featureToggleInitialState,
+    FeaturetoggleContext,
+} from '../../ducks/feature-toggles';
+import userEvent from '@testing-library/user-event';
+
+type ProviderProps = {
+    brukerregistrering?: DeepPartial<BrukerRegistreringData>;
```

Comment:
Hva er forskjellen fra den rå typen BrukerRegistreringData her?

Reply:
Med den faktiske typen, vil typescript kreve at alle verdier er satt i henhold til typen. Men når vi bruker en `Partial<>`, kan vi definere kun de propertiene som er relevante. Siden vi kjører en `merge` på denne verdien på toppen av `initialState`, vil vi få (halv-)fornuftige verdier for alt annet som ikke eksplisitt defineres her.

Dette gjør at vi kan skrive f.eks.
```typescript
const providerProps: ProviderProps = {
  …
  situasjon: { opprettet: '2019-04-07T11:53:05.486686+01:00' },
}
```
uten at den forventer at alle feltene i `situasjon` er med. Datoen er det eneste relevante for testen, så jeg synes det blir ryddigere å kun vise datoen.

`DeepPartial<>` er det samme som `Partial<>`, men at den støtter partials rekursivt. Altså kan vi sende inn del-verdier i flere nivåer ned, bl.a. som vi gjør i `situasjon`-eksempelet over.